content/blog/managing-google-workspace-with-terraform.md

```diff
@@ -0,0 +1,295 @@
+
+1. **Import an existing Google Workspace**  
+   We shared the [terraform code](https://github.com/masterpointio/terraform-googleworkspace-users-groups-automation/blob/main/examples/import-existing-org/main.tf) we used to declare existing [users](https://github.com/masterpointio/terraform-googleworkspace-users-groups-automation/blob/main/examples/import-existing-org/users.yaml) and [groups](https://github.com/masterpointio/terraform-googleworkspace-users-groups-automation/blob/main/examples/import-existing-org/groups.yaml) in YAML, and the [import blocks](https://github.com/masterpointio/terraform-googleworkspace-users-groups-automation/blob/main/examples/import-existing-org/imports.tf) to easily import an existing setup. We additionally shared a [Python script](https://github.com/masterpointio/terraform-googleworkspace-users-groups-automation/blob/main/examples/import-existing-org/debugging-script.py) to view the JSON of your existing resources (which you could give to an LLM and tell it to follow the example to create your own `users.yaml` and `groups.yaml` files)..
+
+2. **Create New Resources**  
+   Start fresh by managing Google Workspace users and groups directly from Terraform. This is a great starting point if you're setting up a new org or moving toward automation-first infra practices.
+
+After the initial import or setup, your Google Workspace [root module](https://newsletter.masterpoint.io/p/multi-instance-vs-single-instance-root-modules) could end up being as simple as this:
+
+```terraform
+locals {
+  oauth_scopes = [
+    "https://www.googleapis.com/auth/admin.directory.group",
+    "https://www.googleapis.com/auth/admin.directory.user",
+    "https://www.googleapis.com/auth/admin.directory.userschema",
+    "https://www.googleapis.com/auth/apps.groups.settings",
+    "https://www.googleapis.com/auth/iam",
+  ]
+}
+
+provider "googleworkspace" {
+  customer_id             = "my_customer"
+  credentials             = local.secrets["googleworkspace_admin_credentials_json"]
+  impersonated_user_email = var.impersonated_user_email
+  oauth_scopes            = local.oauth_scopes
+}
+
+# Get users and groups from YAML files. You might choose to use other config file.
+# Note: path.module refers to the directory where terraform/tofu plan/apply is run
+# var.googleworkspace_configs is a variable for the directory containing config files
+locals {
+  config_path = "${path.module}/${var.googleworkspace_configs}"
+  all_groups = yamldecode(file("${local.config_path}/groups.yaml"))
+  all_users  = yamldecode(file("${local.config_path}/users.yaml"))
+
+  # Skip objects that start with "_", which we use as default or prototype objects
+  groups = { for k, v in local.all_groups : k => v if !startswith(k, "_") }
+  users  = { for k, v in local.all_users : k => v if !startswith(k, "_") }
+}
+
+module "googleworkspace_users_groups" {
+  source  = "masterpointio/users-groups-automation/googleworkspace"
+  version = "X.X.X"
```

Comment:
does this mean "specify a version, but we aren't because this is example code?" That's what I took it to mean.